crates/router/src/connector/globalpay/transformers.rs

```diff
@@ -0,0 +1,188 @@
+use error_stack::ResultExt;
+use serde::{Deserialize, Serialize};
+
+use super::{
+    requests::{self, GlobalpayPaymentsRequest},
+    response::{GlobalpayPaymentStatus, GlobalpayPaymentsResponse},
+};
+use crate::{
+    connector::utils::{self, CardData, PaymentsRequestData},
+    core::errors,
+    types::{self, api, storage::enums},
+    utils::OptionExt,
+};
+
+impl TryFrom<&types::PaymentsAuthorizeRouterData> for GlobalpayPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(item: &types::PaymentsAuthorizeRouterData) -> Result<Self, Self::Error> {
+        let metadata = item
+            .connector_meta_data
+            .to_owned()
+            .get_required_value("connector_meta_data")
+            .change_context(errors::ConnectorError::NoConnectorMetaData)?;
+        let account_name = metadata
+            .as_object()
+            .and_then(|o| o.get("account_name"))
+            .map(|o| o.to_string())
+            .ok_or_else(utils::missing_field_err("connector_meta.account_name"))?;
+        let card = item.get_card()?;
+        Ok(Self {
+            account_name,
+            amount: item.request.amount.to_string(),
+            currency: item.request.currency.to_string(),
+            reference: item.get_attempt_id()?,
+            country: item.get_billing_country()?,
+            capture_mode: item.request.capture_method.map(|f| match f {
+                enums::CaptureMethod::Manual => requests::CaptureMode::Later,
+                _ => requests::CaptureMode::Auto,
+            }),
+            payment_method: requests::PaymentMethod {
+                card: Some(requests::Card {
+                    number: card.get_card_number(),
+                    expiry_month: card.get_card_expiry_month(),
+                    expiry_year: card.get_card_expiry_year_2_digit(),
+                    cvv: card.get_card_cvc(),
+                    ..Default::default()
+                }),
+                ..Default::default()
+            },
+            ..Default::default()
+        })
+    }
+}
+
+impl TryFrom<&types::PaymentsCaptureRouterData> for GlobalpayPaymentsRequest {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(value: &types::PaymentsCaptureRouterData) -> Result<Self, Self::Error> {
+        Ok(Self {
+            amount: match value.request.amount_to_capture {
+                Some(amount) => amount.to_string(),
+                _ => "".to_string(), // takes the original amount of the transaction to capture
```

Comment:
Is this a todo or the connector takes the transaction amount as capture amount?

If connector takes  transaction amount means this should be `None`?

Reply:
It not a todo. we are using the same payment request struct for capture also, it is mandatory for authorize but not mandatory for capture, so passing "" will solve that purpose instead of making it optional